kubernetes/coreos/kubernetes-cluster.yaml

```diff
@@ -0,0 +1,369 @@
+
+    - name: etcd2.service
+      command: start
+
+    - name: weave-network.target
+      enable: true
+      content: |
+        [Unit]
+        Description=Weave Network Setup Complete
+        Documentation=man:systemd.special(7)
+        RefuseManualStart=no
+        After=network-online.target
+        [Install]
+        WantedBy=multi-user.target
+        WantedBy=kubernetes-master.target
+        WantedBy=kubernetes-node.target
+
+    - name: kubernetes-master.target
+      enable: true
+      command: start
+      content: |
+        [Unit]
+        Description=Kubernetes Cluster Master
+        Documentation=http://kubernetes.io/
+        RefuseManualStart=no
+        After=weave-network.target
+        Requires=weave-network.target
+        ConditionHost=kube-01
+        Wants=kube-apiserver.service
+        Wants=kube-scheduler.service
+        Wants=kube-controller-manager.service
+        Wants=kube-proxy.service
+        [Install]
+        WantedBy=multi-user.target
+
+    - name: kubernetes-node.target
+      enable: true
+      command: start
+      content: |
+        [Unit]
+        Description=Kubernetes Cluster Node
+        Documentation=http://kubernetes.io/
+        RefuseManualStart=no
+        After=weave-network.target
+        Requires=weave-network.target
+        ConditionHost=!kube-01
+        Wants=kube-proxy.service
+        Wants=kubelet.service
+        [Install]
+        WantedBy=multi-user.target
+
+    - name: 10-weave.network
+      runtime: false
+      content: |
+        [Match]
+        Type=bridge
+        Name=weave*
+        [Network]
+
+    - name: install-weave.service
+      enable: true
+      content: |
+        [Unit]
+        After=network-online.target
+        After=docker.service
+        Before=weave.service
+        Description=Install Weave
+        Documentation=http://docs.weave.works/
+        Requires=network-online.target
+        [Service]
+        Type=oneshot
+        RemainAfterExit=yes
+        ExecStartPre=/bin/mkdir -p /opt/bin/
+        ExecStartPre=/usr/bin/curl \
+          --silent \
+          --location \
+          https://raw.github.com/weaveworks/weave/master/weave \
+          --output /opt/bin/weave
+        ExecStartPre=/usr/bin/chmod +x /opt/bin/weave
+        ExecStartPre=/bin/bash -c 'test ! -f /tmp/weave.tar || docker load -i /tmp/weave.tar'
+        ExecStart=/bin/echo Weave Installed
+        [Install]
+        WantedBy=weave-network.target
+        WantedBy=weave.service
+
+    - name: weaveproxy.service
+      enable: true
+      content: |
+        [Unit]
+        After=install-weave.service
+        After=docker.service
+        Description=Weave Network Router
+        Documentation=http://docs.weave.works/
+        Requires=docker.service
+        Requires=install-weave.service
+        [Service]
+        EnvironmentFile=-/etc/weave.env
+        ExecStartPre=/opt/bin/weave launch-proxy --rewrite-inspect --without-dns
```

Comment:
That one used to be necessary, can be changed now.

On Sun, 6 Sep 2015 08:43 Matthias Radestock <notifications@github.com>
wrote:

> In kubernetes/coreos/kubernetes-cluster.yaml
> <https://github.com/weaveworks/guides/pull/52#discussion_r38818579>:
>
> > +        WantedBy=weave-network.target
> > +        WantedBy=weave.service
> > +
> > +    - name: weaveproxy.service
> > +      enable: true
> > +      content: |
> > +        [Unit]
> > +        After=install-weave.service
> > +        After=docker.service
> > +        Description=Weave Network Router
> > +        Documentation=http://docs.weave.works/
> > +        Requires=docker.service
> > +        Requires=install-weave.service
> > +        [Service]
> > +        EnvironmentFile=-/etc/weave.env
> > +        ExecStartPre=/opt/bin/weave launch-proxy --rewrite-inspect --without-dns
>
> why is there no weave setup here, as there is for weave.service?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/weaveworks/guides/pull/52/files#r38818579>.
>